src/Microsoft.AspNetCore.Sockets.Client.Http/Internal/TaskExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Sockets.Http.Internal
+{
+    public static class NoThrowTaskExtensions
+    {
+        public static NoThrowAwaiter NoThrowInternal(this Task task)
+        {
+            return new NoThrowAwaiter(task);
+        }
+        public static async Task NoThrow(this Task task)
+        {
+            await new NoThrowAwaiter(task);
+        }
+    }
+
+    public struct NoThrowAwaiter : ICriticalNotifyCompletion
```

Comment:
I don't think it should be public. Either make it a private class or move to a different file.